packages/round-manager/src/features/api/utils.ts

```diff
@@ -176,37 +176,35 @@ export const abbreviateAddress = (address: string) => `${address.slice(0, 8)}...
  * @param metadata - The metadata of a round application
  * @returns The application schema
  */
-export const generateApplicationSchema = (metadata: any) => {
-
-  // declare schema with default fields
-  let schema = [];
-
-  for (const key of Object.keys(metadata)) {
-    if (typeof metadata[key] === "object") {
-
-      for (const subKey of Object.keys(metadata[key])) {
-        schema.push({
-          id: schema.length,
-          question: camelToTitle(subKey),
-          type: "TEXT",
-          required: true,
-          info: metadata[key][subKey],
-          choices: []
-        })
-      }
-      continue
-
-    } else {
-      schema.push({
-        id: schema.length,
-        question: camelToTitle(key),
-        type: "TEXT",
-        required: true,
-        info: metadata[key],
-        choices: []
-      })
-    }
+export const generateApplicationSchema = (metadata: ApplicationMetadata) => {
+  if (!metadata.customQuestions) return []
+
+  const maybe_questions = metadata.customQuestions
+
+  const isPresent = (item: (string | undefined)[]): item is string[] => {
+    return !!item[1]
   }
 
-  return schema
+  const questions = [
+    ["email", maybe_questions.email],
+    ["twitter", maybe_questions.twitter],
+    ["website", maybe_questions.website],
+    ["github", maybe_questions.github],
+    ["githubOrganization", maybe_questions.githubOrganization],
+    ["fundingSource", maybe_questions.fundingSource],
+    ["profit2022", maybe_questions.profit2022],
+    ["teamSize", maybe_questions.teamSize]
+  ].filter(isPresent)
+
+  return questions.map(([name , answer], i) => {
+    return {
+      id: i,
+      question: camelToTitle(name),
+      type: "TEXT",
+      required: true,
+      info: answer,
+      choices: [],
+      encrypted: name === 'email'
```

Comment:
worth adding a TODO comment that we would need to make remove this hardcoding and make it dynamic ?

Reply:
Personally, I don't much mind the static nature of the type checking here. I guess in this case it's all strings and wild-westy, but my preference would be to take the hit of having to update this when the input changes as a way of keeping ourselves honest about what we add to the metadata. 

Either way I'll add the TODO